### PR TITLE
refactor: Creates new module for crash reporting

### PR DESCRIPTION
**Pull request by Kaytbode**
A new file report.ts that handles all codes dealing with crash
reporting and sentry init, was created in the extension directory.

The changes did not cause any side effects after running tests.

### CODE REVIEW

**Comment by ryanluker** [comment]
@Kaytbode thanks for the PR, to get the tests passing again I will need to merge my other one first then these should be working again. https://github.com/ryanluker/vscode-coverage-gutters/pull/283/

Once that happens please update your branch and then I will start a PR review for yah.
Thanks once again for the contribution! It helps a ton keeping this project active and maintained for all 🦾 .

**Comment by Kaytbode** [comment]
> @Kaytbode thanks for the PR.
> To get the tests passing, I will need to merge my other one first then these should be working again. #283
> 
> Once that happens please update your branch and then I will start a PR review for yah.
> Also, thanks once again for the contribution! It helps a ton keeping this project active and maintained for all 🦾 .

Ok. I appreciate the opportunity to be contributing to such a wonderful project.  My pleasure

**Comment by Kaytbode** [comment]
@ryanluker  Please take a look at my code. I cannot figure out why the tests are failing here, but passing on my local machine.

**Comment by Kaytbode** [comment]
@ryanluker  I have resolved it, by enabling TSLINT on my code editor.

File: src/extension/gutters.ts
Comment (ryanluker) [inline src/extension/gutters.ts:116]: instead of using capture error to figure out if crash reporting is on how about we make `checkCrashReporterEnabled` public and we can check that way?

File: src/extension/gutters.ts
Comment (ryanluker) [inline src/extension/gutters.ts:118]: we will want to move `crashReporter.captureError(area, error);` inside this if check to keep all the sentry bits in the same logically area.

File: src/extension/report.ts
Comment (ryanluker) [inline src/extension/report.ts:8]: we can make this a boolean type and have it default to false I think (for safety).

File: src/extension/report.ts
Comment (ryanluker) [inline src/extension/report.ts:13]: should we also figure out if crash reporting is on at this point?
then we can store that on the instance of the reporter instead
```typescript
        const telemetry = workspace.getConfiguration("telemetry");
        this.enableCrashReporting = telemetry.get("enableCrashReporter");
```

File: src/extension/report.ts
Comment (ryanluker) [inline src/extension/report.ts:11]: not sure we need anything passed into the constructor as we always use `telemetry` to figure out if we want to activate crash reporting 🤔

File: src/extension/report.ts
Comment (ryanluker) [inline src/extension/report.ts:16]: I think we can add a return type of `string | undefined` ?

File: src/extension/report.ts
Comment (ryanluker) [inline src/extension/report.ts:42]: I think this block of code makes more sense in the constructor and thus only able to instantiated once?
Currently we could call manualCapture many times and potentially init sentry many times as well 🤔 .

File: src/extension/report.ts
Comment (ryanluker) [inline src/extension/report.ts:7]: I would update the name of the file to be `crashreporter.ts` to make discoverability easier.

**Review by ryanluker — CHANGES_REQUESTED** [review]
Looking pretty great so far! Mostly just some code design changes and adding the crash reporter in a few more places now that we have the `CrashReporter` class.

**Extra places:**
1) `private handleError(system: string, error: Error)` in `coverageparser.ts`
2) `private handleError(error: Error)` in `coverageservice.ts`

File: src/extension/report.ts
Comment (Kaytbode) [reply src/extension/report.ts:42]: Ok.. will work on it

File: src/extension/report.ts
Comment (Kaytbode) [reply src/extension/report.ts:16]: I thought about that too. Ok will change it.

File: src/extension/report.ts
Comment (Kaytbode) [reply src/extension/report.ts:11]: Ok. That is right

**Comment by Kaytbode** [comment]
@ryanluker Please take a look at my code again. I have effected the changes you mentioned.

File: src/extension/report.ts
Comment (Kaytbode) [reply src/extension/report.ts]: I returned an array of string instead, to cater for the sentryId and sentryPrompt.

**Review by ryanluker — APPROVED** [review]
Thanks again for the contribution, this is looking great!
I can do some QA after this is merged to the mainline branch and it will then go out in 2.7.0

File: src/extension/crashreporter.ts
Comment (Kaytbode) [reply src/extension/crashreporter.ts:7]: Thank you

File: src/coverage-system/coverageservice.ts
Comment (Kaytbode) [reply src/coverage-system/coverageservice.ts:187]: > Thanks again for the contribution, this is looking great!
> I can do some QA after this is merged to the mainline branch and it will then go out in 2.7.0

My pleasure. Thank you for guiding me through it

**Comment by ryanluker** [comment]
@Kaytbode I can merge this in for you but once I gave the approval you should have been able to merge to the mainline branch as well?

**Comment by Kaytbode** [comment]
> @Kaytbode I can merge this in for you but once I gave the approval you should have been able to merge to the mainline branch as well?

I was able to merge to my copy of the mainline branch. Are you saying I can merge to ryanluker/vscode-coverage-gutters directly?

**Comment by ryanluker** [comment]
@Kaytbode I guess that was the bit that you weren't able to do even with the pull request being approved? I don't mind pushing the button but I was just curious 😁 .

**Comment by Kaytbode** [comment]
> @Kaytbode I guess you weren't able to merge to `ryanluker/vscode-coverage-gutters` even with the pull request being approved?
> I don't mind pushing the button but I was just curious 😁 .

Oh ok.. I did not even try to...

**Pull request merged**
